File: src/main/java/com/xhb/pay/bean/PayInsideCarOnlinePayTempOrder.java
package com.xhb.pay.bean;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fhs.core.base.bean.BaseDO;
import com.fhs.core.group.Add;
import com.fhs.core.group.Update;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;

/**
 * 内部客户网络支付订单(PayInsideCarOnlinePayTempOrder)实体类
 *
 * @author makejava
 * @since 2019-05-25 15:38:01
 */

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@TableName("t_pay_inside_car_online_pay_temp_order")
public class PayInsideCarOnlinePayTempOrder extends BaseDO<PayInsideCarOnlinePayTempOrder> {
    private static final long serialVersionUID = 287019896948588371L;
    @TableId(value = "id", type = IdType.INPUT)
    private Long id;

    //停车场id
    @TableField("park_id")
    private Long parkId;

    //前段用户id
    @Length(message = "前段用户id字段的长度最大为32", groups = {Add.class, Update.class}, max = 32)
    @TableField("front_user_id")
    private String frontUserId;

    //物业费
    @TableField("property_fee")
    private Double propertyFee;

    //0 未缴纳 1 已缴纳 2 已退款
    @TableField("property_fee_status")
    private Integer propertyFeeStatus;

    //物业费gateway订单号
    @Length(message = "物业费gateway订单号字段的长度最大为32", groups = {Add.class, Update.class}, max = 32)
    @TableField("property_fee_order_no")
    private String propertyFeeOrderNo;

    //车位数量
    @TableField("lot_num")
    private Integer lotNum;

    //开始生效日期
    @Length(message = "开始生效日期字段的长度最大为10", groups = {Add.class, Update.class}, max = 10)
    @TableField("start_date")
    private String startDate;

    //结束生效日期
    @Length(message = "结束生效日期字段的长度最大为10", groups = {Add.class, Update.class}, max = 10)
    @TableField("end_date")
    private String endDate;

    //停车费费用
    @TableField("monthly_fee")
    private Double monthlyFee;

    //0 未缴纳 1已缴纳 2已退款
    @TableField("monthly_fee_status")
    private Integer monthlyFeeStatus;

    //停车费订单号
    @Length(message = "停车费订单号字段的长度最大为32", groups = {Add.class, Update.class}, max = 32)
    @TableField("monthly_fee_order_no")
    private String monthlyFeeOrderNo;

    //物业费支付时间
    @Length(message = "物业费支付时间字段的长度最大为20", groups = {Add.class, Update.class}, max = 20)
    @TableField("property_fee_pay_time")
    private String propertyFeePayTime;

    //物业费退款时间
    @Length(message = "物业费退款时间字段的长度最大为20", groups = {Add.class, Update.class}, max = 20)
    @TableField("property_fee_return_time")
    private String propertyFeeReturnTime;

    //停车费付款时间
    @Length(message = "停车费付款时间字段的长度最大为20", groups = {Add.class, Update.class}, max = 20)
    @TableField("monthly_fee_pay_time")
    private String monthlyFeePayTime;

    //停车费退款时间
    @Length(message = "停车费退款时间字段的长度最大为20", groups = {Add.class, Update.class}, max = 20)
    @TableField("monthly_fee_return_time")
    private String monthlyFeeReturnTime;

    //车位ids
    @Length(message = "车位ids字段的长度最大为1024", groups = {Add.class, Update.class}, max = 1024)
    @TableField("lot_ids")
    private String lotIds;

    //计费规则id
    @TableField("monthly_rule_id")
    private Long monthlyRuleId;

    //内部客户id
    @TableField("inside_id")
    private Long insideId;

    //月租户交费记录id
    @TableField("contract_id")
    private Long contractId;

    //超时时间
    @Length(message = "超时时间字段的长度最大为20", groups = {Add.class, Update.class}, max = 20)
    @TableField("time_out_time")
    private String timeOutTime;

    @Length(message = "${column.comment}字段的长度最大为32", groups = {Add.class, Update.class}, max = 32)
    @TableField("group_code")
    private String groupCode;

    //几个月
    @TableField("month_num")
    private Double monthNum;

    //用户姓名
    @Length(message = "用户姓名字段的长度最大为32", groups = {Add.class, Update.class}, max = 32)
    @TableField("name")
    private String name;

    //手机号
    @Length(message = "手机号字段的长度最大为20", groups = {Add.class, Update.class}, max = 20)
    @TableField("mobile")
    private String mobile;

    //用“，”分割的车位号字符串
    @TableField(exist = false)
    private String lotNos;


}

File: src/main/webapp/pagex/pay/pay_inside_car_online_pay_temp_order.js
var modelConfig= {title:'网络月租订单',pkey:'id',type:'snow', orderBy:'update_time Desc',
    namespace:"pay_inside_car_online_pay_temp_order",table:'t_pay_inside_car_online_pay_temp_order',trans:true,
    isMultiTenant:true,dp:JSON.stringify({park_id:'parkIds'}),exjs:'${path.staticPath}/js/My97DatePicker/WdatePicker.js',
};
var listPage={
    listFieldSett:function(){
        return [
            {name:'park_id',title:'停车场名称',width:'20%',align:'center',trans:'simple',target:'com.xhb.park.bean.ParkParking',field:'parkName',showField:'transMap.parkName'},
            {name: 'name',title:'用户姓名',width:'10%',align:'center'},
            {name: 'mobile',title:'手机号',width:'10%',align:'center'},
            /*{name: 'property_fee',title:'物业费用',width:'10%',align:'center'},
            {name: 'property_fee_status',title: '物业费用状态',width: '10%',align: 'center',trans: 'book',key: 'fee_status',showField: 'transMap.property_fee_statusName'},*/
            {name: 'lot_num',title:'车位数量',width:'10%',align:'center'},
            {name: 'monthly_fee',title:'停车费费用',width:'10%',align:'center'},
            {name: 'monthly_fee_status',title: '停车费用状态',width: '10%',align: 'center',trans: 'book',key: 'fee_status',showField: 'transMap.monthly_fee_statusName'},
            {name:'create_time',title:'创建时间',width:'15%',align:'center'},
            {name: 'start_date',title:'开始生效日期',width:'10%',align:'center'},
            {name: 'end_date',title:'结束生效日期',width:'9%',align:'center'},
        ]},
    isColumnButton:function(){
        return  true;
    },
    filters:function(){
        return [
            {name:'park_id',type:'select',url:'${path.basePath}/ms/x/parking/findListData',valuefield:'id',textfield:'parkName',title:'停车场',showAll:true},
            {name:'name',type:'input',title:'姓名',filterType:'like'},
            {name:'mobile',type:'input',title:'电话',filterType:'like'},
            /*{name:'property_fee_status',title:'物业费费用状态',type:'book',code:'fee_status',showAll:true},*/
            {name:'monthly_fee_status',title:'停车费费用状态',type:'book',code:'fee_status',showAll:true},
            {name: 'create_time', type: 'dateBT', title: '创建时间',isBT:true},
        ];
    },
    buttons:function(){
        return [
        ];
    },
    disableButtons:function(){
        return ["add","update","delete"];
    },
    otherFunctions:function(){
        return {
        }
    }
};

var add={
    formFields:function(){
        return [
            {name: 'park_id',type:'select',url:'${path.basePath}/ms/x/parking/findListData',valuefield:'id',textfield:'parkName',title:'停车场名称',required:true},
            {name: 'name', title: '用户姓名' ,type:'input'},
            {name: 'mobile', title: '手机号' ,type:'input'},
            {name: 'property_fee', title: '物业费' ,type:'input'},
            {name: 'property_fee_status', title: '物业费费用状态', type: 'book', code: 'fee_status'},
            {name: 'property_fee_order_no', title: '物业费订单号' ,type:'input'},
            {name: 'lot_num', title: '车位数量' ,type:'input'},
            {name: 'start_date', title: '开始生效日期' ,type:'input'},
            {name: 'end_date', title: '结束生效日期' ,type:'input'},
            {name: 'monthly_fee', title: '停车费' ,type:'input'},
            {name: 'monthly_fee_status', title: '停车费费用状态', type: 'book', code: 'fee_status'},
            {name: 'monthly_fee_order_no', title: '停车费订单号' ,type:'input'},
            /*{name: 'property_fee_pay_time', title: '物业费支付时间' ,type:'input'},
            {name: 'property_fee_return_time', title: '物业费退款时间' ,type:'input'},*/
            {name: 'monthly_fee_pay_time', title: '停车费付款时间' ,type:'input'},
            {name: 'monthly_fee_return_time', title: '停车费退款时间' ,type:'input'},
            {name: 'lot_ids', type:'select',url:'${path.basePath}/report/parkRegionLot/getLotNOAndRegionName', valuefield:'id',textfield:'lotNo',title:'车位',multiple:true},
            {name: 'monthly_rule_id',title:'计费规则',type:'select',url:'${path.basePath}/ms/x/park_monthly_rule/findListData',valuefield:'id',textfield:'ruleName',required:true},
            {name: 'inside_id',title:'内部客户',type:'select',url:'${path.basePath}/ms/x/pay_inside_car/findListData',valuefield:'id',textfield:'ownerName',required:true},
            {name: 'time_out_time', title: '超时时间' ,type:'input'},
            {name: 'month_num', title: '月数' ,type:'input'},
            {name: 'create_time',title:'创建时间',type:'input'},
        ];
    },
    otherFunctions:function(){
        return {
            ready:function(){
            },
            loadSuccess:function(info){

            },
            onSave:function(){
            },
            saveSucess:function(){
            },
            saveError:function(){

            },
        }
    }
}
